build.rs

```diff
@@ -70,3 +73,46 @@ _app_count:
     }
     Ok(())
 }
+
+fn gen_kernel_config(arch: &str) -> Result<()> {
+    let config_path = PathBuf::from(format!("configs/{}.toml", arch));
+    let config = std::fs::read_to_string(config_path)?;
+    let out_path = PathBuf::from(std::env::var("OUT_DIR").unwrap()).join("uspace_config.rs");
+    let config_content = config
+        .parse::<DocumentMut>()
+        .expect("failed to parse config file")
+        .as_table()
+        .clone();
+
+    fn get_comments<'a>(config: &'a Table, key: &str) -> &'a str {
+        config
+            .key(key)
+            .and_then(|k| k.leaf_decor().prefix())
+            .and_then(|s| s.as_str())
+            .map(|s| s.trim())
+            .unwrap_or_default()
+    }
+
+    let mut f = File::create(out_path)?;
+    writeln!(f, "// Automatically generated by build.rs\n")?;
+    for (key, item) in config_content.iter() {
+        let comments = get_comments(&config_content, key).replace('#', "///");
+        writeln!(f, "{}", comments)?;
+        if let Item::Value(value) = item {
+            let key_name = key.to_uppercase().replace('-', "_");
+            writeln!(f, "#[allow(dead_code)]")?;
```

Comment:
`#[allow(dead_code)]` -> `#![allow(dead_code)]`, and move  to the line between 97 and 98.